### DISCUSSION TITLE
Release Candidate 10

### DISCUSSION THREAD

**Discussion by t1stm** [t1stm/ThirtyDollarTools#25] [Q&A]
# More Custom Events And Better Performance??? At The Price Of An Update!
I guess I really like the 2 month gaps between updates, huh? Anyways let's get to business. 
This update features things I've wanted to see implemented in the TDW one way or another, where if people like them they might get implemented in the real site too.

I'll only cover the new events added, since I doubt anyone reading this will want to hear me talk about multi-threading and other task separation stuff. I'll now start with the two new events.

## The #icut event.
This event behaves just like the normal `|!cut|` event but with a twist. It cuts only the given sounds. Here's an example.

```|boom@0.7%99|!combine|mariopaint_flower|_pause|#icut(boom, mariopaint_flower)|```
```|#icut(sound1, sound2, sound3, sound_and_so_on)|```

## The "^" modifier.
Time for some backstory. While implementing the #icut event, I started making some testing covers that used it and one of those was of the OST track, "Another Medium" in UNDERTALE. One issue was that it relied heavily on instrument separation between the left and right channels, which the TDW doesn't support at the moment of writing this. And so, I implemented this too since it was easy to add, and in my opinion something that should be added to the TDW some day. 

The modifier works like this:
`|mariopaint_gameboy^0.45|# Pans so that the right channel is 45% louder than the left.|`
`|mariopaint_gameboy^-0.45|# Pans so that the left channel is 45% louder than the right.|`

I'll now show a snippet of an instrument track in my cover, which you can check out in the Visualizer or Converter.

```
# Another Medium snippet, covered by t1stm.|

#define(square_wave)|
mariopaint_gameboy@0%100|
#enddefine|

!speed@130|!speed@4@x|!divider|

square_wave@-3^0.45|
square_wave@-6^0.45|
square_wave@6|
square_wave@1|

square_wave@13^-0.45|
square_wave@1^-0.45|
square_wave@4^0.45|
square_wave@1^0.45|

square_wave@11|
square_wave@4|
square_wave@6^-0.45|
square_wave@-1^-0.45|

square_wave@2^0.45|
square_wave@-1^0.45|
square_wave@6|
square_wave@2|!divider|

square_wave@-1^-0.45|
square_wave@-6^-0.45|
square_wave@2^0.45|
square_wave@-1^0.45|

square_wave@13|
square_wave@6|
square_wave@14^-0.45|
square_wave@9^-0.45|

square_wave@16^0.45|
square_wave@9^0.45|
square_wave@14|
square_wave@7|

square_wave@13^-0.45|
square_wave@6^-0.45|
square_wave@9^0.45|
square_wave@2^0.45|!divider|
```

## That's it.
Well that's it for this update, as I don't have a lot of stuff to cover. This was mostly a back end cleaning type of update but it'll help with a lot of things in the future. For now I've started experimenting with new ways to visualize covers, so keep an eye on the releases tab at least once a month if you're interest. I'll see you in the next release!

## What's Changed (too many commits for me to cover. look at the pull requests or full changelog instead.)
* Update the Thirty Dollar Visualizer to take advantage of multiple threads. by @t1stm in https://github.com/t1stm/ThirtyDollarTools/pull/23
* Add custom events and performance improvements. by @t1stm in https://github.com/t1stm/ThirtyDollarTools/pull/24

**Full Changelog**: https://github.com/t1stm/ThirtyDollarTools/compare/v1.0.0-rc9-3...v1.0.0-rc10

<hr /><em>This discussion was created from the release <a href='https://github.com/t1stm/ThirtyDollarTools/releases/tag/v1.0.0-rc10'>Release Candidate 10</a>.</em>

**Comment by KaloEagle** [comment]
Very slow process, I feel displeased. I hope the dev fixes it. :)

**Reply by t1stm** [reply]
I know about this issue and it will be fixed in the final 1.0 release, which is coming soon.
